packages/@magic-sdk/provider/src/modules/user.ts

```diff
@@ -47,19 +51,59 @@ export class UserModule extends BaseModule {
   }
 
   public isLoggedIn() {
-    const requestPayload = createJsonRpcRequestPayload(
-      this.sdk.testMode ? MagicPayloadMethod.IsLoggedInTestMode : MagicPayloadMethod.IsLoggedIn,
-    );
-    return this.request<boolean>(requestPayload);
+    return createPromiEvent<boolean, any>(async (resolve, reject) => {
+      try {
+        let cachedIsLoggedIn = false;
+        if (this.sdk.useStorageCache) {
+          cachedIsLoggedIn = (await getItem(this.localForageIsLoggedInKey)) === 'true';
+
+          // if isLoggedIn is true on storage, optimistically resolve with true
+          // if it is false, we use `usr.isLoggedIn` as the source of truth.
+          if (cachedIsLoggedIn) {
+            resolve(true);
```

Comment:
IIRC, using resolve without return won't actually end the function execution. To prevent further code execution, it's better to use return resolve(true), as this will bypass any additional calls that follow.

Reply:
That's actually intended, because we need to do the actual call to `isLoggedIn` after resolving `true` when `cachedIsLoggedIn` is `true`. That way we can emit `emitUserLoggedOut` in the case of a false positive, as explained [here](https://fortmatic.slack.com/archives/CSS9C804V/p1703104490572939?thread_ts=1703102678.495479&cid=CSS9C804V).

In other words, we're optimistically resolving `true` to gain speed by avoiding the call to `isLoggedIn`, but we still do the actual call to `isLoggedIn` to make sure the user is actually logged in.

Reply:
Sounds good to me!